File: Hanoi.py
def hanoi(N, origen='1', dest='3', aux='2'):

    if N > 0:
        hanoi(N-1, origen, aux, dest)
        print ('Se mueve de torre', origen, 'a torre', dest)
        hanoi(N-1, aux, dest, origen)

if __name__ == '__main__':
    discos = int(input("Ingresa el numero de discos: "))
    hanoi(discos)

'''
hanoi ( N, origen, destino, auxiliar )
{
si N mayor a 0 entonces
hanoi(N - 1, origen, auxiliar, destino)
pasar disco N de origen a destino
hanoi(N - 1, auxiliar, destino, origen)
fin-si
}
'''

File: Hanoi.java
package hanoi;

import java.util.Scanner;

public class Hanoi{

    static Scanner sc = new Scanner(System.in);
    
    static void hanoi(int N, int origen, int dest, int aux){
        
        if(N > 0){
            hanoi(N-1, origen, aux, dest);
            System.out.println("Se mueve de torre " + origen + " a torre " + dest);
            hanoi(N-1, aux, dest, origen);
        }
    }
    
    public static void main(String[] args){
        
        System.out.println("Ingresa el numero de discos: ");
        int N = sc.nextInt();
        hanoi(N, 1, 3, 2);
    }
}
